apps/web/src/services/iam/components/UserGroupChannelDeleteDoubleCheckModal.vue

```diff
@@ -19,28 +19,33 @@ const userGroupPageStore = useUserGroupPageStore();
 const userGroupPageState = userGroupPageStore.state;
 const userGroupPageGetters = userGroupPageStore.getters;
 
+const queryClient = useQueryClient();
+const { userGroupChannelAPI } = useUserGroupChannelApi();
+const { key: userGroupChannelListQueryKey } = useServiceQueryKey('alert-manager', 'user-group-channel', 'list');
+
 const emit = defineEmits<{(e: 'confirm'): void; }>();
 
-const state = reactive({
-    loading: false,
+const { mutate: userGroupChannelDeleteMutate, isPending: userGroupChannelDeletePending } = useMutation({
+    mutationFn: (params: UserGroupChannelDeleteParameters) => userGroupChannelAPI.delete(params),
+    onSuccess: () => {
+        emit('confirm');
+        showSuccessMessage('', i18n.t('IAM.USER_GROUP.MODAL.DELETE.SHOW_SUCCESS_MESSAGE'));
+        userGroupPageState.userGroupChannels.selectedIndices = [];
+        queryClient.invalidateQueries({ queryKey: userGroupChannelListQueryKey.value });
+    },
+    onError: (error) => {
+        ErrorHandler.handleError(error, true);
+    },
+    onSettled: () => {
+        handleCancel();
+    },
 });
 
 /* Component */
-const handleConfirm = async () => {
-    try {
-        state.loading = true;
-        if (userGroupPageGetters.selectedUserGroups && userGroupPageGetters.selectedUserGroups[0].notification_channel && userGroupPageGetters.selectedUserGroups[0].notification_channel.length > 0) {
-            await fetchDeleteNotificationChannel({
-                channel_id: userGroupPageGetters.selectedUserGroups[0].notification_channel[userGroupPageState.userGroupChannels.selectedIndices[0]].channel_id,
-            });
-            emit('confirm');
-            showSuccessMessage('', i18n.t('IAM.USER_GROUP.MODAL.DELETE.SHOW_SUCCESS_MESSAGE'));
-            userGroupPageState.userGroupChannels.selectedIndices = [];
-        }
-    } finally {
-        state.loading = false;
-        handleCancel();
-    }
+const handleConfirm = () => {
+    userGroupChannelDeleteMutate({
+        channel_id: userGroupPageGetters.selectedUserGroupChannel[0].channel_id,
+    });
```

Comment:
Suggestion: I think this needs defensive code~!

such as
```suggestion
const handleConfirm = () => {
    if (!userGroupPageGetters.selectedUserGroupChannel?.[0]?.channel_id) {
        console.warn('blah blah');
        return;
    }
    userGroupChannelDeleteMutate({
        channel_id: userGroupPageGetters.selectedUserGroupChannel[0].channel_id,
    });
    
```